From 266a090882133e30df20899bbf8a5b66b28e02cd Mon Sep 17 00:00:00 2001
From: OPNA2608 <opna2608@protonmail.com>
Date: Mon, 14 Oct 2024 00:31:01 +0200
Subject: [PATCH] Disable update checking

Downloading an AppImage and trying to run it just won't work.
---
 src/qt_gui/check_update.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/qt_gui/check_update.cpp b/src/qt_gui/check_update.cpp
index ca6009ca..e3b14d5d 100644
--- a/src/qt_gui/check_update.cpp
+++ b/src/qt_gui/check_update.cpp
@@ -201,6 +201,12 @@ void CheckUpdate::setupUI(const QString& downloadUrl, const QString& latestDate,
     noButton->setSizePolicy(QSizePolicy::Minimum, QSizePolicy::Preferred);
     bottomLayout->addWidget(autoUpdateCheckBox);
 
+    yesButton->setVisible(false);
+    yesButton->setEnabled(false);
+    QString updateDisabledText = QStringLiteral("[Nix] Auto-updating has been disabled in this package.");
+    QLabel* updateDisabledLabel = new QLabel(updateDisabledText, this);
+    layout->addWidget(updateDisabledLabel);
+
     QSpacerItem* spacer = new QSpacerItem(40, 20, QSizePolicy::Expanding, QSizePolicy::Minimum);
     bottomLayout->addItem(spacer);
 
-- 
2.44.1

